### PR TITLE
Add PlatformBrightness hook

### PR DESCRIPTION
**Pull request by AhmedLSayed9**

### CODE REVIEW

File: packages/flutter_hooks/lib/src/platform_brightness.dart [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+typedef BrightnessCallback = FutureOr<void> Function(
+  Brightness? brightness,
+);
+
+/// Returns the current Platform Brightness value and rebuilds the widget when it changes.
+Brightness? usePlatformBrightness() {
+  return use(const _PlatformBrightnessHook(rebuildOnChange: true));
+}
+
+useOnPlatformBrightnessChange(BrightnessCallback? onBrightnessChange) {
```
Comment (rrousselGit) [inline packages/flutter_hooks/lib/src/platform_brightness.dart:13]: Type as `void`

File: packages/flutter_hooks/lib/src/platform_brightness.dart [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  final BrightnessCallback? onBrightnessChange;
+
+  @override
+  _PlatformBrightnessState createState() => _PlatformBrightnessState();
+}
+
+class _PlatformBrightnessState
+    extends HookState<Brightness?, _PlatformBrightnessHook>
+    with WidgetsBindingObserver {
+  Brightness? _brightness;
```
Comment (rrousselGit) [inline packages/flutter_hooks/lib/src/platform_brightness.dart:33]: Should be non-nullable, using `late`

platformBrightness is never null, and WidgetsBinding.instance should never be null either

File: packages/flutter_hooks/lib/src/platform_brightness.dart [hunk 3]
```diff
@@ -0,0 +1,60 @@
+part of 'hooks.dart';
+
+/// A callback triggered when the platform brightness changes.
+typedef BrightnessCallback = FutureOr<void> Function(
+  Brightness? brightness,
+);
+
+/// Returns the current Platform Brightness value and rebuilds the widget when it changes.
+Brightness? usePlatformBrightness() {
```
Comment (rrousselGit) [inline packages/flutter_hooks/lib/src/platform_brightness.dart:9]: Should return a non-nullable value

File: packages/flutter_hooks/lib/src/platform_brightness.dart [hunk 4]
```diff
@@ -0,0 +1,60 @@
+part of 'hooks.dart';
+
+/// A callback triggered when the platform brightness changes.
+typedef BrightnessCallback = FutureOr<void> Function(
+  Brightness? brightness,
```
Comment (rrousselGit) [inline packages/flutter_hooks/lib/src/platform_brightness.dart:5]: Should be non-nullable

Comment (rrousselGit) [reply packages/flutter_hooks/lib/src/platform_brightness.dart:5]: We probably also want to pass the previous value

**Comment by rrousselGit** [comment]
Hello! Thanks for this!

Could you add tests for this?

**Comment by rrousselGit** [comment]
And also documentation (dart doc, readme, changelog)

File: packages/flutter_hooks/lib/src/platform_brightness.dart (hunk 1 above)
Comment (rrousselGit) [inline packages/flutter_hooks/lib/src/platform_brightness.dart:13]: Needs documentation

File: packages/flutter_hooks/lib/src/platform_brightness.dart [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+  final bool rebuildOnChange;
+  final BrightnessCallback? onBrightnessChange;
+
+  @override
+  _PlatformBrightnessState createState() => _PlatformBrightnessState();
+}
+
+class _PlatformBrightnessState
+    extends HookState<Brightness?, _PlatformBrightnessHook>
```
Comment (rrousselGit) [inline packages/flutter_hooks/lib/src/platform_brightness.dart:31]: Needs to override debugLabel/debugValue  
See https://github.com/rrousselGit/flutter_hooks/blob/c87b6d947b3b5cb1b1b210f121ce752b767da748/packages/flutter_hooks/lib/src/async.dart#L114

**Comment by AhmedLSayed9** [comment]
Thank you for the instructions, All done.

File: packages/flutter_hooks/test/use_platform_brightness_test.dart [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+import 'package:flutter/widgets.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+
+import 'mock.dart';
+
+void main() {
+  group('usePlatformBrightness', () {
+    testWidgets('returns initial value and rebuild widgets on change',
+        (tester) async {
+      tester.binding.window.platformBrightnessTestValue = Brightness.light;
```
Comment (rrousselGit) [inline packages/flutter_hooks/test/use_platform_brightness_test.dart:12]: It seems like `.window.platformBrightnessTestValue` is deprecated. Could you update the tests to use the new solution?

**Comment by rrousselGit** [comment]
LGTM once the deprecated values are updated.

Thanks for your work!

File: packages/flutter_hooks/test/use_platform_brightness_test.dart (hunk 6 above)
Comment (AhmedLSayed9) [reply packages/flutter_hooks/test/use_platform_brightness_test.dart:12]: the new solution is `.platformDispatcher.platformBrightnessTestValue` but it's not on Stable channel yet.
Should I switch to beta channel and update it?

Comment (AhmedLSayed9) [reply packages/flutter_hooks/test/use_platform_brightness_test.dart:12]: @rrousselGit

Comment (rrousselGit) [reply packages/flutter_hooks/test/use_platform_brightness_test.dart:12]: Sorry for the delay. Yes, please do so. There should be a new stable release soon

Make sure to update the Flutter SDK constraints in the pubspec.yaml too

File: packages/flutter_hooks/CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,7 @@
+## 0.18.3
+
+- Added `usePlatformBrightness` and `useOnPlatformBrightnessChange` to interact with platform `Brightness`
```
Comment (rrousselGit) [inline packages/flutter_hooks/CHANGELOG.md:3]: Could you fix the merge conflicts?

**Comment by rrousselGit** [comment]
LGTM, thanks!

**Pull request merged**